stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/feature/messages/list/adapter/view/internal/PollView.kt

```diff
@@ -0,0 +1,273 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.ui.feature.messages.list.adapter.view.internal
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import androidx.viewbinding.ViewBinding
+import io.getstream.chat.android.models.Option
+import io.getstream.chat.android.models.Poll
+import io.getstream.chat.android.models.Vote
+import io.getstream.chat.android.models.VotingVisibility
+import io.getstream.chat.android.ui.databinding.StreamUiItemPollAnswerBinding
+import io.getstream.chat.android.ui.databinding.StreamUiItemPollCloseBinding
+import io.getstream.chat.android.ui.databinding.StreamUiItemPollHeaderBinding
+import io.getstream.chat.android.ui.databinding.StreamUiItemPollResultsBinding
+import io.getstream.chat.android.ui.utils.extensions.createStreamThemeWrapper
+import io.getstream.chat.android.ui.utils.extensions.streamThemeInflater
+import io.getstream.log.taggedLogger
+
+internal class PollView : RecyclerView {
+
+    private val logger by taggedLogger("PollView")
+
+    // private lateinit var style: FileAttachmentViewStyle
+
+    private lateinit var pollAdapter: PollAdapter
+    var onOptionClick: ((Option) -> Unit) = { _ -> }
+    var onClosePollClick: ((Poll) -> Unit) = { _ -> }
+    var onViewPollResultsClick: ((Poll) -> Unit) = { _ -> }
+
+    constructor(context: Context) : this(context, null, 0)
+
+    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)
+
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
+        context.createStreamThemeWrapper(),
+        attrs,
+        defStyleAttr,
+    ) {
+        init(attrs)
+    }
+
+    init {
+        layoutManager = LinearLayoutManager(context)
+        itemAnimator = null
+    }
+
+    private fun init(attrs: AttributeSet?) {
+        /** not implemented yet */
+    }
+
+    fun setPoll(poll: Poll, isMine: Boolean) {
+        logger.d { "[setPoll] poll: $poll" }
+        if (!::pollAdapter.isInitialized) {
+            pollAdapter = PollAdapter(
+                onOptionClick = { option -> onOptionClick(option) },
+                onClosePollClick = { onClosePollClick(poll) },
+                onViewPollResultsClick = { onViewPollResultsClick(poll) },
+            )
+            adapter = pollAdapter
+        }
+
+        val pollItems = mutableListOf<PollItem>()
+        pollItems.add(PollItem.Header(poll.name, poll.description))
+        println("JcLog [setPoll]: poll.voteCountsByOption -> ${poll.voteCountsByOption}")
+        println("JcLog [setPoll]: poll.votes.count -> ${poll.votes.size}")
```

Comment:
Maybe we need to use the logger here or remove it?

Reply:
Good catch.
They were only debug logs during development that are not needed at all